nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -284,6 +315,33 @@ NavfnPlanner::makePlan(
     // extract the plan
     if (getPlanFromPotential(best_pose, plan)) {
       smoothApproachToGoal(best_pose, plan);
+
+      // Override last pose orientation to match goal if use_final_approach_orientation=false
+      // (default).
+      // If use_final_approach_orientation=true, interpolate the last pose orientation from the
+      // previous pose.
+      // And deal with corner case of plan of length 1
+      size_t plan_size = plan.poses.size();
+      if (plan_size == 1) {
```

Comment:
hum, then we need to nest everything in a `if (plan_size > 0)` (because we don't want to interfere with the planner returning a null path):

>       if (plan_size > 0){
>         if (use_final_approach_orientation_) {
>           
>         } else { 
>           
>         }
>       }

or to avoid adding a block, but less clearer :

>       if (use_final_approach_orientation_ && plan_size > 0) {
>         
>       } else if (!use_final_approach_orientation_ && plan_size > 0){ 
>         
>       }

What do you prefer

Reply:
That's fair, I wasn't super explicit

```
if (use_final_approach_orientation_) {
  size_t plan_size = plan.poses.size();
  if (plan_size == 1) {
    plan.poses.back().pose.orientation = start.orientation;
  } else if (plan_size > 1) {
    double dx, dy, theta;
    auto last_pose = plan.poses.back().pose.position;
    auto approach_pose = plan.poses[plan_size - 2].pose.position;
    // Deal with the case of NavFn producing a path with two equal last poses
    if (std::abs(last_pose.x - approach_pose.x) < 0.0001 &&
      std::abs(last_pose.y - approach_pose.y) < 0.0001 && plan_size > 2)
    {
      approach_pose = plan.poses[plan_size - 3].pose.position;
    }
    dx = last_pose.x - approach_pose.x;
    dy = last_pose.y - approach_pose.y;
    theta = atan2(dy, dx);
    plan.poses.back().pose.orientation =
      nav2_util::geometry_utils::orientationAroundZAxis(theta);
  }
```

Reply:
I think that's alot more easily readable, no?

Reply:
Definitly, thx!

Reply:
I remove the override of the goal orientation in NavFn and Theta*, only needed for Smac2d and modify the comments to reflect it